include/alpaka/rand/RandPhilox.hpp

```diff
@@ -0,0 +1,232 @@
+             * @param seed Set the Philox generator key
+             * @param subsequence Select a subsequence of size 2^64
+             * @param offset Number of numbers to skip form the start of the subsequence.
+             */
+            ALPAKA_FN_HOST_ACC Philox4x32x10Vector(
+                std::uint32_t const seed = 0,
+                std::uint32_t const subsequence = 0,
+                std::uint32_t const offset = 0)
+                : engineVariant(seed, subsequence, offset)
+            {
+            }
+
+            ALPAKA_FN_HOST_ACC Philox4x32x10Vector(EngineVariant state) : engineVariant(state)
+            {
+            }
+
+            EngineVariant engineVariant;
+            template<typename TScalar>
+            using ResultContainer = typename EngineVariant::template ResultContainer<TScalar>;
+
+            using ResultInt = std::uint32_t;
+            using ResultVec = decltype(engineVariant());
+            ALPAKA_FN_HOST_ACC constexpr ResultInt min()
+            {
+                return 0;
+            }
+            ALPAKA_FN_HOST_ACC constexpr ResultInt max()
+            {
+                return std::numeric_limits<ResultInt>::max();
+            }
+            ALPAKA_FN_HOST_ACC ResultVec operator()()
+            {
+                return engineVariant();
+            }
+            // TODO: https://en.cppreference.com/w/cpp/named_req/RandomNumberEngine
+        };
+
+
+        // The following exists because you "cannot call __device__ function from a __host__ __device__ function"
+        // directly, but wrapping that call in a struct is just fine.
+        template<typename TEngine>
+        struct EngineCallHostAccProxy
+        {
+            ALPAKA_FN_HOST_ACC auto operator()(TEngine& engine) -> decltype(engine())
+            {
+                return engine();
+            }
+        };
+
+        /// TEMP: Distributions to be decided on later. The generator should be compatible with STL as of now.
+        template<typename TResult, typename TSFINAE = void>
+        class UniformReal
+        {
+        public:
+            UniformReal() = delete;
+        };
+
+        /// TEMP: Distributions to be decided on later. The generator should be compatible with STL as of now.
+        template<typename TResult>
+        class UniformReal<TResult, std::enable_if_t<meta::IsArrayOrVector<TResult>::value>>
+            : public concepts::Implements<ConceptRand, UniformReal<TResult>>
+        {
+            using T = typename TResult::value_type;
+
+        public:
+            ALPAKA_FN_HOST_ACC UniformReal() : UniformReal(0, 1)
+            {
+            }
+
+            ALPAKA_FN_HOST_ACC UniformReal(T min, T max) : _min(min), _max(max), _range(_max - _min)
+            {
+            }
+
+            template<typename TEngine>
+            ALPAKA_FN_HOST_ACC TResult operator()(TEngine& engine)
+            {
+                // FIXME: quick hack for 4-vectors, to be changed after API decision.
```

Comment:
So has the API decision been taken yet? Do we still need this FIXME?

Reply:
No decision has been made - we barely started a discussion :-) I'll remove the comment and create a github issue instead